lib/helper.js

```diff
@@ -98,12 +98,66 @@ const commandlineArgs = () => {
     return args;
 };
 
+const sleep = milliseconds => {
+    var start = new Date().getTime();
+    for (var i = 0; i < 1e7; i++)
+        if ((new Date().getTime() - start) > milliseconds) break;
+};
+
+const exists = get => {
+    return async () => {
+        if ((await get()).length) return true;
+        return false;
+    };
+};
+
+const getIfExists = get => {
+    return async (tag, intervalTime = 1000, timeout = 10000) => {
+        try {
+            await waitUntil(async () => (await get(tag)).length > 0, intervalTime, timeout);
+            return await get(tag);
+        } catch (e) {
+            return [];
+        }
+    };
+};
+
+const waitUntil = async (condition, intervalTime, timeout) => {
+    var start = new Date().getTime();
+    while (true) {
+        try {
+            if (await condition()) break;
+        } catch (e) { }
+        if ((new Date().getTime() - start) > timeout)
+            throw new Error(`waiting failed: timeout ${timeout}ms exceeded`);
+        sleep(intervalTime);
+    }
+};
+
+const waitForNavigation = (timeout, promises = []) => {
+    return new Promise((resolve, reject) => {
+        Promise.all(promises).then(resolve);
+        const timeoutId = setTimeout(() => reject('Timedout'), timeout);
+        timeouts.push(timeoutId);
```

Comment:
this timeout has to be cleared on closebrowser https://github.com/getgauge/taiko/blob/ca5933931bab5b273dd558abf92fc68a8c0da9f8/lib/taiko.js#L164

Reply:
Having generic names like `utilities.js`, `common.js` and 'helpers.js` is an anti-pattern as they become a dump yard of functions not related to each other. 

Not in the scope of this PR but at a later point we must consider splitting files according to category or domains.